### PR TITLE
Expose additional Chai-1 parameters in the pipeline

### PR DESCRIPTION
**Pull request by FloWuenne**
- Added params `num_trunk_recycles`, `num_diffn_timesteps`, `seed` and `use_esm_embeddings` so they can be modified via Nextflow
- Update nextflow_schema.json and hide these new parameters in Platform

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
                     "type": "string",
                     "format": "directory-path",
                     "description": "The output directory where the results will be saved. You have to use absolute paths to storage on Cloud infrastructure.",
-                    "fa_icon": "fas fa-folder-open"
+                    "fa_icon": "fas fa-folder-open",
+                    "help_text": ""
```
Comment (drpatelh) [inline nextflow_schema.json]:
```suggestion
                    "fa_icon": "fas fa-folder-open"
```

File: nextflow.config [hunk 2]
```diff
@@ -13,6 +13,10 @@ params {
     input                        = null
     weights_dir                  = null
     use_gpus                     = false
+    num_trunk_recycles          = 3
+    num_diffn_timesteps         = 200
+    seed                        = 42
+    use_esm_embeddings           = true
```
Comment (drpatelh) [inline nextflow.config]:
```suggestion
    num_trunk_recycles           = 3
    num_diffusion_timesteps      = 200
    seed                         = 42
    use_esm_embeddings           = true
```

File: nextflow.config [hunk 3]
```diff
@@ -13,6 +13,10 @@ params {
     input                        = null
     weights_dir                  = null
     use_gpus                     = false
+    num_trunk_recycles          = 3
+    num_diffn_timesteps         = 200
```
Comment (drpatelh) [inline nextflow.config:17]:
```suggestion
    num_diffusion_timesteps     = 200
```

File: modules/local/chai_1/main.nf [hunk 4]
```diff
@@ -7,25 +7,32 @@ process CHAI_1 {
...
+    def esm_flag = use_esm_embeddings ? '--use-esm-embeddings' : ''
     """
-    CHAI_DOWNLOADS_DIR=$downloads_dir \\
     run_chai_1.py \\
-        --output-dir ${meta.id} \\
-        --fasta-file ${fasta}
+        --fasta-file ${fasta} \\
+        --output-dir . \\
+        --num-trunk-recycles ${num_trunk_recycles} \\
+        --num-diffn-timesteps ${num_diffn_timesteps} \\
```
Comment (drpatelh) [inline modules/local/chai_1/main.nf]:
```suggestion
        --num-diffn-timesteps ${num_diffusion_timesteps} \\
```

File: modules/local/chai_1/main.nf [hunk 5]
```diff
@@ -7,25 +7,32 @@ process CHAI_1 {
...
+    val use_esm_embeddings
 
     output:
     tuple val(meta), path("${meta.id}/*.cif"), emit: structures
     tuple val(meta), path("${meta.id}/*.npz"), emit: arrays
     path "versions.yml"                      , emit: versions
 
     script:
-    def downloads_dir = weights_dir ?: './downloads'
+    def args = task.ext.args ?: ''
```
Comment (drpatelh) [inline modules/local/chai_1/main.nf]:
```suggestion
```

File: modules/local/chai_1/main.nf [hunk 6]
```diff
@@ -7,25 +7,32 @@ process CHAI_1 {
     input:
     tuple val(meta), path(fasta)
     path weights_dir
+    val num_trunk_recycles
+    val num_diffn_timesteps
```
Comment (drpatelh) [inline modules/local/chai_1/main.nf]:
```suggestion
    val num_diffusion_timesteps
```

File: bin/run_chai_1.py [hunk 7]
```diff
@@ -40,11 +65,11 @@ def main():
     run_inference(
         fasta_file=args.fasta_file,
         output_dir=args.output_dir,
-        num_trunk_recycles=3,
-        num_diffn_timesteps=200,
-        seed=42,
+        num_trunk_recycles=args.num_trunk_recycles,
+        num_diffn_timesteps=args.num_diffn_timesteps,
```
Comment (drpatelh) [inline bin/run_chai_1.py:69]:
```suggestion
        num_diffn_timesteps=args.num_diffusion_timesteps,
```

File: modules/local/chai_1/main.nf [hunk 8]
```diff
@@ -7,25 +7,32 @@ process CHAI_1 {
...
+        --output-dir . \\
+        --num-trunk-recycles ${num_trunk_recycles} \\
+        --num-diffn-timesteps ${num_diffn_timesteps} \\
+        --seed ${seed} \\
+        ${esm_flag} \\
+        $args
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
-        python: \$(python --version | sed 's/Python //g')
```
Comment (drpatelh) [inline modules/local/chai_1/main.nf:26]: Why did you remove this @FloWuenne ?

File: modules/local/chai_1/main.nf [hunk 9]
```diff
@@ -7,25 +7,32 @@ process CHAI_1 {
...
         chai_lab: \$(python -c "import chai_lab; print(chai_lab.__version__)")
-        torch: \$(python -c "import torch; print(torch.__version__)")
```
Comment (drpatelh) [inline modules/local/chai_1/main.nf:28]: Why did you remove this @FloWuenne ?

File: nextflow_schema.json [hunk 10]
```diff
@@ -39,6 +40,34 @@
...
                     "description": "Run compatible tasks on GPUs rather than CPUs (default).",
                     "fa_icon": "fas fa-microchip"
+                },
+                "num_trunk_recycles": {
+                    "type": "integer",
+                    "default": 3,
+                    "fa_icon": "fas fa-recycle",
+                    "hidden": true
+                },
+                "num_diffn_timesteps": {
```
Comment (drpatelh) [inline nextflow_schema.json]:
```suggestion
                "num_diffusion_timesteps": {
```

File: nextflow_schema.json [hunk 11]
```diff
@@ -39,6 +40,34 @@
...
+                    "hidden": true,
+                    "description": "Number of diffusion steps to use."
+                },
+                "seed": {
+                    "type": "integer",
+                    "default": 42,
+                    "fa_icon": "fas fa-seedling",
+                    "hidden": true,
+                    "description": "Random seed to be used for Chai-1 calculations",
+                    "help_text": ""
```
Comment (drpatelh) [inline nextflow_schema.json]:
```suggestion
                    "description": "Random seed to be used for Chai-1 calculations"
```

File: workflows/nf_chai/main.nf [hunk 12]
```diff
@@ -34,7 +34,11 @@ workflow NF_CHAI {
     // Run structure prediction with Chai-1
     CHAI_1 (
         ch_fasta,
-        weights_dir ? Channel.fromPath(weights_dir) : []
+        weights_dir ? Channel.fromPath(weights_dir) : [],
+        params.num_trunk_recycles,
+        params.num_diffn_timesteps,
+        params.seed,
+        params.use_esm_embeddings
```
Comment (drpatelh) [inline workflows/nf_chai/main.nf]:
```suggestion
        num_trunk_recycles,
        num_diffusion_timesteps,
        seed,
        use_esm_embeddings
```

File: nextflow_schema.json [hunk 13]
```diff
@@ -39,6 +40,34 @@
...
+                },
+                "use_esm_embeddings": {
+                    "type": "boolean",
+                    "default": true,
+                    "fa_icon": "fas fa-stamp",
+                    "hidden": true,
+                    "description": "Use user-provided esm model embeddings?"
```
Comment (drpatelh) [inline nextflow_schema.json]:
```suggestion
                    "description": "Use user-provided esm model embeddings"
```

File: nextflow_schema.json [hunk 14]
```diff
@@ -39,6 +40,34 @@
                     "type": "boolean",
                     "description": "Run compatible tasks on GPUs rather than CPUs (default).",
                     "fa_icon": "fas fa-microchip"
+                },
+                "num_trunk_recycles": {
+                    "type": "integer",
+                    "default": 3,
+                    "fa_icon": "fas fa-recycle",
```
Comment (drpatelh) [inline nextflow_schema.json:46]:
```suggestion
                    "fa_icon": "fas fa-recycle",
                    "description": "Number of trunk recycles",
```

File: workflows/nf_chai/main.nf [hunk 15]
```diff
@@ -16,8 +16,8 @@ include { CHAI_1                 } from '../../modules/local/chai_1'
 workflow NF_CHAI {
 
     take:
-    fasta_file  //  string: path to fasta file read provided via --input parameter
-    weights_dir //  string: path to model directory read provided via --weights_directory parameter
+    fasta_file            //  string: path to fasta file read provided via --input parameter
+    weights_dir           //  string: path to model directory read provided via --weights_directory parameter
```
Comment (drpatelh) [inline workflows/nf_chai/main.nf]:
```suggestion
    fasta_file              //  string: path to fasta file read provided via --input parameter
    weights_dir             //  string: path to model directory read provided via --weights_directory parameter
    num_trunk_recycles      // integer: Number of trunk recycles
    num_diffusion_timesteps // integer: Number of diffusion steps to use
    seed                    // integer: Random seed to be used for Chai-1 calculations
    use_esm_embeddings      // boolean: Use user-provided esm model embeddings
```

File: modules/local/chai_1/main.nf (hunk 8 above)
Comment (FloWuenne) [reply modules/local/chai_1/main.nf:26]: Don't remember removing this, must have been accidental, sorry!
